### PR TITLE
Add sample_weight_name argument to Reductions

### LINKED ISSUE #580: Can't use sklearn.Pipeline with GridSearch.
Hello,

I may be wrong, but GridSearch does not allow [sklearn.Pipeline](https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html) for its estimator.
This would be a useful feature have!

## How to reproduce.
In `test/unit/reductions/grid_search/test_grid_search_arguments.py` add:

```python
# Additional Imports
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import StandardScaler

# Set up Pipeline estimator
class TestPipelineEstimator(ConditionalOpportunityTests):
    def setup_method(self, method):
        self.estimator = Pipeline([('scaler', StandardScaler()), ('logistic', LogisticRegression(solver='liblinear'))])
        self.disparity_criterion = DemographicParity()
```

Then : `pytest test/unit/reductions/grid_search/test_grid_search_arguments.py` would fail with:

```
ValueError: Pipeline.fit does not accept the sample_weight parameter. You can pass parameters to specific steps of your pipeline using the stepname__parameter format, e.g. `Pipeline.fit(X, y, logisticregression__sample_weight=sample_weight)`.
```

### Proposed solution

In GridSearch constructor, we could allow specifying the key for `sample_weight`.

Ex. `GridSearch(self.estimator, self.disparity_criterion, grid_size=2, sample_weight_key='logistic__sample_weight')`

I already made the test so I can add the solution too if it is approved.

### PR DESCRIPTION
**Pull request by Dref360**
Allow the user to supply sample_weight_key to better control the call to estimator.fit.

Fixes #580

### CODE REVIEW

File: fairlearn/reductions/_exponentiated_gradient/_lagrangian.py [hunk 1]
```diff
@@ -38,9 +38,12 @@ class _Lagrangian:
     opt_lambda : bool
         indicates whether to optimize lambda during the calculation of the
         Lagrangian; optional with default value True
+    sample_weight_key : str
+        Argument to supply `sample_weight` in `estimator.fit`.
```
Comment (riedgar-ms) [inline fairlearn/reductions/_exponentiated_gradient/_lagrangian.py:42]: Can I suggest "Name of the argument to `estimator.fit()` which supplies the sample weights (defaults to `sample_weight`)"

File: fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py [hunk 2]
```diff
@@ -51,17 +51,21 @@ class ExponentiatedGradient(BaseEstimator, MetaEstimatorMixin):
         if True each step of exponentiated gradient is followed by the saddle
         point optimization over the convex hull of classifiers returned so
         far; default True
+    sample_weight_key : str
+        Argument to supply `sample_weight` in `estimator.fit`.
```
Comment (riedgar-ms) [inline fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py:55]: See above for suggested change

**Review by riedgar-ms — COMMENTED** [review]
Should be reviewed by @MiroDudik  too.

**Review by riedgar-ms — APPROVED** [review]

File: fairlearn/reductions/_exponentiated_gradient/_lagrangian.py (hunk 1 above)
Comment (MiroDudik) [reply fairlearn/reductions/_exponentiated_gradient/_lagrangian.py]: I think we should change `sample_weight_key` to `sample_weight_name`.

File: fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py (hunk 2 above)
Comment (MiroDudik) [reply fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py]: Similarly, I think we should change `sample_weight_key` to `sample_weight_name`.

**Review by romanlutz — APPROVED** [review]
I'm very excited to get these improvements in! There's nothing I want changed in the code, but you should add your name to the AUTHORS.md and add a short description to CHANGES.md

**Comment by Dref360** [comment]
I added my name and a small description. :)

**Review by MiroDudik — APPROVED** [review]

**Pull request merged**